egg/core/callbacks.py

```diff
@@ -159,6 +164,8 @@ def save_checkpoint(self, filename: str):
         Saves the game, agents, and optimizer states to the checkpointing path under `<number_of_epochs>.tar` name
         """
         self.checkpoint_path.mkdir(exist_ok=True, parents=True)
+        if len(self.get_checkpoint_files()) > self.max_checkpoints:
```

Comment:
This should be done only if `max_checkpoints > 0`. If you think at the very corner case where you are resuming a training, by mistake (maybe forgot to update your code) you set the checkpoint path and have `max_checkpoints = 0` then you'd be removing old checkpoints from the previous run

Reply:
Well yes, if the number of chkps is grater than max_chkps then the old one must be removed

Reply:
This is very very minor though, you can ignore it for the sake of the PR, up to you

Reply:
It depends if you consider ma_checkpoints per run (then you shouldn't change them) or max_checkpoints per folder? I think we should max per run. If I set max_checkpoints to 0 it should mean I don't want any checkpoint to be saved in this run, the expected behavior is not deleting checkpoints from a previous run

Let's leave it as is for now though